drivers/bridge/bridge.go

```diff
@@ -1451,3 +1479,60 @@ func electMacAddress(epConfig *endpointConfiguration, ip net.IP) net.HardwareAdd
 	}
 	return netutils.GenerateMACFromIP(ip)
 }
+
+func (d *driver) RestoreBridgeEndpoint(n *bridgeNetwork, endpoint *bridgeEndpoint, iface driverapi.InterfaceInfo) error {
+	ipv4Addr := iface.Address()
+	// Restore ipv4
+	_, err := ipAllocator.RequestIP(n.bridge.bridgeIPv4, ipv4Addr.IP)
+	if err != nil {
+		return err
+	}
+
+	n.Lock()
+	config := n.config
+	n.Unlock()
+
+	// Restore ipv6
+	ipv6Addr := iface.AddressIPv6()
+	if ipv6Addr.IP != nil && len(ipv6Addr.IP) != 0 {
+		network := n.bridge.bridgeIPv6
+		if config.FixedCIDRv6 != nil {
+			network = config.FixedCIDRv6
+		}
+		_, err := ipAllocator.RequestIP(network, ipv6Addr.IP)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Restore container interface name
+	sbox, err := linkByMacAddress(iface.MacAddress().String())
+	if err != nil {
+		return err
```

Comment:
Testing the final changes by building a docker binary. It failed with the following error when I tried to restore a bunch of saved endpoints.

```
ERRO[0000] failed to create endpoint s3 on network network1: Link with hardware addr 02:42:ac:19:00:01 not found
ERRO[0000] failed to create endpoint s4 on network network1: Link with hardware addr 02:42:ac:19:00:02 not found
ERRO[0000] failed to create endpoint c1 on network bridge: Link with hardware addr 02:42:ac:11:00:02 not found
ERRO[0000] failed to create endpoint s5 on network network2: Link with hardware addr 02:42:ac:1a:00:01 not found
```